doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md

```diff
@@ -0,0 +1,582 @@
+# NAT
+Management Support for NAT Feature
+
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 10/30/2019  |   Ravi Vasantahm, Arthi Sivanantham      | Initial version                   |
+
+# About this Manual
+
+This document provides north bound interface details for NAT feature.
+
+# Scope
+
+This document covers the management interfaces supported for NAT feature and Unit test cases. It does not include the protocol design or protocol implementation details.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| NAT                      | Network Address Translation  |
+
+# 1 Feature Overview
+
+Provide CLI, GNMI and REST management framework capabilities for CONFIG and GET support for NAT feature.
+
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide CLI, REST and gNMI support for configuring and displaying NAT attributes.
+
+
+### 1.1.2 Configuration and Management Requirements
+
+Supported Commands via:
+- CLI style commands
+- REST API support
+- gNMI Support
+
+### 1.1.3 Scalability Requirements
+N/A
+
+### 1.1.4 Warm Boot Requirements
+N/A
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+Will be enhancing the management framework backend and transformer methods to add support for NAT configuration and display.
+
+
+### 1.2.2 Container
+All code changes will be done in management-framework container.
+
+### 1.2.3 SAI Overview
+N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+N/A
+## 2.2 Functional Description
+N/A
+
+
+# 3 Design
+## 3.1 Overview
+N/A
+## 3.2 DB Changes
+No changes to existing DBs and no new DB being added.
+### 3.2.1 CONFIG DB
+### 3.2.2 APP DB
+
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+N/A
+### 3.3.1 Orchestration Agent
+N/A
+### 3.3.2 Other Process
+N/A
+
+## 3.4 SyncD
+N/A
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+
+IETF NAT Yang (RFC 8512) is used for the north bound management interface support.
```

Comment:
So, now the IETF Yang is not used for the north bound management interface support, right?

Reply:
Yes, not going to use IETF yang. Will be defining openconfig style yang for NAT, will update HLD once yang model is ready.